sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -57,14 +57,14 @@ object CollateExpressionBuilder extends ExpressionBuilder {
     expressions match {
       case Seq(e: Expression, collationExpr: Expression) =>
         (collationExpr.dataType, collationExpr.foldable) match {
-          case (StringType, true) =>
+          case (_: StringType, true) =>
```

Comment:
This seems like it was a problem which was created long before default collation. Could we update the name of the PR to something that expresses this. I believe it was not possible to collate any expression which did not evaluate to UTF8_BINARY.

Reply:
Could be nice to add a test with some expression as well, for example `Substring` or any you particularly like.

Reply:
Updated the name. 
Regarding the test, it will be added as part of https://github.com/apache/spark/pull/46561 which originally caught this issue.